### PR TITLE
[Xamarin.MacDev] Add an AppleSdkVersion struct which replaces IPhoneSdkVersion and MacOSXSdkVersion.

### PR DESCRIPTION
**Pull request by rolfbjarne**
This reduces code duplication and makes it possible to simplify logic in a few places.

### CODE REVIEW

**Review by dalexsoto — APPROVED** [review]

File: Xamarin.MacDev/MacOSXSdk.cs [hunk 1]
```diff
@@ -32,15 +32,15 @@ namespace Xamarin.MacDev
...
-            		MacOSXSdkVersion.V10_14
+		List<AppleSdkVersion> knownOSVersions = new List<AppleSdkVersion> {
+			AppleSdkVersion.V10_7,
+			AppleSdkVersion.V10_8,
+			AppleSdkVersion.V10_9,
+			AppleSdkVersion.V10_10,
+			AppleSdkVersion.V10_11,
+			AppleSdkVersion.V10_12,
+			AppleSdkVersion.V10_13,
+			AppleSdkVersion.V10_14,
```
Comment (spouliot) [inline Xamarin.MacDev/MacOSXSdk.cs:43]: not sure who use the API but should include 10.15, 11.[0-3] ?
Also not so clear about the new minor version of 11.x

File: Xamarin.MacDev/MonoTouchSdk.cs [hunk 2]
```diff
@@ -33,42 +33,42 @@ namespace Xamarin.MacDev
...
+			AppleSdkVersion.V8_1,
+			AppleSdkVersion.V8_2,
+			AppleSdkVersion.V8_3,
+			AppleSdkVersion.V8_4,
+			AppleSdkVersion.V9_0,
+			AppleSdkVersion.V9_1,
+			AppleSdkVersion.V9_2,
+			AppleSdkVersion.V9_3,
+			AppleSdkVersion.V10_0,
+			AppleSdkVersion.V10_1
```
Comment (spouliot) [inline Xamarin.MacDev/MonoTouchSdk.cs:51]: same comment, that seems incomplete
who's using this ?

File: Xamarin.MacDev/XamMacSdk.cs [hunk 3]
```diff
@@ -16,13 +16,13 @@ namespace Xamarin.MacDev
...
-			MacOSXSdkVersion.V10_10,
-			MacOSXSdkVersion.V10_11,
-			MacOSXSdkVersion.V10_12
+		static readonly AppleSdkVersion [] MacOSXSdkVersions = {
+			AppleSdkVersion.V10_7,
+			AppleSdkVersion.V10_8,
+			AppleSdkVersion.V10_9,
+			AppleSdkVersion.V10_10,
+			AppleSdkVersion.V10_11,
+			AppleSdkVersion.V10_12
```
Comment (spouliot) [inline Xamarin.MacDev/XamMacSdk.cs:25]: more missing versions
should not it be identical to the previous one ?

**Review by spouliot — APPROVED** [review]
comments are not related to the renaming itself

File: Xamarin.MacDev/MacOSXSdk.cs (hunk 1 above)
Comment (rolfbjarne) [reply Xamarin.MacDev/MacOSXSdk.cs:43]: I believe VSMac are using these.

I didn't want to remove any that VSMac might be using (nor add any VSMac isn't using).

File: Xamarin.MacDev/MonoTouchSdk.cs (hunk 2 above)
Comment (rolfbjarne) [reply Xamarin.MacDev/MonoTouchSdk.cs:51]: Same: VSMac.

File: Xamarin.MacDev/XamMacSdk.cs (hunk 3 above)
Comment (rolfbjarne) [reply Xamarin.MacDev/XamMacSdk.cs:25]: Same.

**Review by emaf — APPROVED** [review]

**Pull request merged**